File: android/src/test/java/com/usercentrics/sdk/flutter/api/FakeUsercentricsBannerProxy.kt
package com.usercentrics.sdk.flutter.api

import com.usercentrics.sdk.BannerSettings
import com.usercentrics.sdk.UsercentricsConsentUserResponse

internal class FakeUsercentricsBannerProxy(
    private val showFirstLayerAnswer: UsercentricsConsentUserResponse? = null,
    private val showSecondLayerAnswer: UsercentricsConsentUserResponse? = null,
) : UsercentricsBannerProxy {

    var showFirstLayerCount = 0
    var showFirstLayerBannerSettingsArgument: BannerSettings? = null

    override fun showFirstLayer(
        bannerSettings: BannerSettings?,
        callback: (UsercentricsConsentUserResponse?) -> Unit
    ) {
        showFirstLayerCount++
        showFirstLayerBannerSettingsArgument = bannerSettings
        callback(showFirstLayerAnswer)
    }

    var showSecondLayerCount = 0
    var showSecondLayerBannerSettingsArgument: BannerSettings? = null

    override fun showSecondLayer(
        bannerSettings: BannerSettings?,
        callback: (UsercentricsConsentUserResponse?) -> Unit
    ) {
        showSecondLayerCount++
        showSecondLayerBannerSettingsArgument = bannerSettings
        callback(showSecondLayerAnswer)
    }

}

File: example/ios/RunnerTests/Fake/FakeUsercentricsBannerProxy.swift
import Foundation
@testable import usercentrics_sdk
@testable import Usercentrics
import UsercentricsUI

class FakeUsercentricsBannerProxy: UsercentricsBannerProxyProtocol {

    var showFirstLayerAnswer: UsercentricsConsentUserResponse? = nil
    var showFirstLayerCount = 0
    var showFirstLayerBannerSettingsArg: BannerSettings? = nil

    func showFirstLayer(bannerSettings: BannerSettings?,
                        completionHandler: @escaping (UsercentricsConsentUserResponse) -> Void) {
        showFirstLayerCount += 1
        showFirstLayerBannerSettingsArg = bannerSettings

        completionHandler(showFirstLayerAnswer!)
    }

    var showSecondLayerAnswer: UsercentricsConsentUserResponse? = nil
    var showSecondLayerCount = 0
    var showSecondLayerBannerSettingsArg: BannerSettings? = nil

    func showSecondLayer(bannerSettings: BannerSettings?,
                         completionHandler: @escaping (UsercentricsConsentUserResponse) -> Void) {
        showSecondLayerCount += 1
        showSecondLayerBannerSettingsArg = bannerSettings

        completionHandler(showSecondLayerAnswer!)
    }

}
